### PR TITLE
Fix custom_scene_viewer for macOS

### PR DESCRIPTION
**Pull request by srmainwaring**
# 🦟 Bug fix

No issue raised but related to #185

## Summary

The example `custom_scene_viewer` does not display a window on macOS. The example will build  and run, the console will display the usage, however the GLUT window does not open.

## Configuration

- OS: macOS Catalina 10.15.7
- Machine: Mac Pro (2019)
- Config: ignition-rendering4 

## Detail

The issue was that the OpenGL context was not managed for macOS. The fix is to add the conditional compilation flags and variables used in the `simple_demo` and adapt them for this example. It was also necessary to move the glutInit() to 
main() as it seems that this is required before the rendering engine is initialised during the scene building.

1. Add OpenGL context settings for macOS to the DemoWindow.
2. Move glutInit to main
3. Add command line option to set the ogre engine.

## Tests

Because of issues with a side-by-side install of ogre and ogre2, tests were limited to ogre1.9 only and ogre2 was brew unlinked before compilation of ignition-rendering.

The images show the first four scenes of the demo running on macOS.

custom_scene_view_1
custom_scene_view_2
custom_scene_view_3
custom_scene_view_4

## Checklist
- [x] Signed all commits for DCO
- [ ] Added tests
- [ ] Updated documentation (as needed)
- [ ] Updated migration guide (as needed)
- [ ] `codecheck` passed (See [contributing](https://ignitionrobotics.org/docs/all/contributing#contributing-code))
- [x] All tests passed (See [test coverage](https://ignitionrobotics.org/docs/all/contributing#test-coverage))
- [ ] While waiting for a review on your PR, please help review [another open pull request](https://github.com/pulls?q=is%3Aopen+is%3Apr+user%3Aignitionrobotics+repo%3Aosrf%2Fsdformat+archived%3Afalse+) to support the maintainers

### CODE REVIEW

**Review by chapulina — COMMENTED** [review]
Thanks for the contribution, we'll be trying it out shortly. Meanwhile, do you mind signing your commit to satisfy the DCO check? See our [contributing guide](https://ignitionrobotics.org/docs/all/contributing#process).

**Comment by srmainwaring** [comment]
Sure - will do. Thought I'd set that up in SourceTree before committing but must have missed something. Are you ok with a force push on the PR once I get the signing in place?

**Comment by chapulina** [comment]
> Are you ok with a force push on the PR once I get the signing in place?

Sure, I think there's no way around this. Thanks!

**Comment by srmainwaring** [comment]
Force pushed signed commit.

**Comment by ahcorde** [comment]
it's still not signed - [DCO](https://github.com/ignitionrobotics/ign-rendering/pull/256/checks?check_run_id=1974527449)

**Comment by srmainwaring** [comment]
Rebased on upstream ign-rendering4 and signed-off

File: examples/custom_scene_viewer/ManualSceneDemo.cc [hunk 1]
```diff
@@ -169,8 +179,18 @@ void ManualSceneDemo::ChangeScene()
...
 //////////////////////////////////////////////////
 //////////////////////////////////////////////////
-int main(int, char**)
+int main(int _argc, char** _argv)
 {
+  glutInit(&_argc, _argv);
+
+  // Expose engine name to command line because we can't instantiate both
+  // ogre and ogre2 at the same time
+  std::string ogre_engine("ogre");
```
Comment (iche033) [inline examples/custom_scene_viewer/ManualSceneDemo.cc:188]: can you change this to camelCase

**Review by iche033 — COMMENTED** [review]
thanks for the patch. Changes look good and consistent with other example code.

Just one minor coding style suggestion.

File: examples/custom_scene_viewer/ManualSceneDemo.cc (hunk 1 above)
Comment (srmainwaring) [reply examples/custom_scene_viewer/ManualSceneDemo.cc]: In 920686c

**Comment by iche033** [comment]
@osrf-jenkins run tests please

**Review by iche033 — APPROVED** [review]

**Pull request merged**